plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,390 @@
+            - For non-ECC keys, this parameter with be ignored.
+        type: str
+        choices: ['cv25519', 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1']
+    key_usage:
+        description:
+            - Specifies usage(s) for key.
+            - Support usages are V(encrypt), V(sign), V(auth), V(cert).
+            - V(cert) is given to all primary keys regardess, however can be used to only give V(vert) usage to a key.
+            - If not usage is specified, the valid usages for the given key type with be assigned.
+            - If O(state) is V(absent), this parameter is ignored. 
+        type: list[str]
+        choices: ['encrypt', 'sign', 'auth', 'cert']
+    subkey_type:
+        description:
+            - Similar to O(key_type), but also supports V(ELG).
+        type: str
+        default: EDDSA
+        choices: ['RSA', 'DSA', 'ECDSA', 'EDDSA', 'ECDH', 'ELG']
+    subkey_length:
+        description:
+            - Similar to O(key_length).
+            - For ELG keys, the minimum is V(1024), the maximum is V(4096), and the default is V(3072)."
+        type: int
+    subkey_curve:
+        description:
+            - "Similar to O(key_curve)"
+        type: str
+        choices: ['cv25519', 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1']
+    key_usage:
+        description:
+            - Similar to O(key_usage), but does not support V(cert).
+        type: list[str]
+        choices: ['encrypt', 'sign', 'auth', 'cert']
+    name:
+        description:
+            - Specifies a name for the key.
+        type: str
+    comment:
+        description:
+            - Specifies a comment for the key.
+        type: str
+    email:
+        description:
+            - Specifies an email for the key.
+        type: str
+    passphrase:
+        description:
+            - Passphrase used to decrypt an existing private key or encrypt a newly generated private key.
+            - If O(state) is V(absent), this parameter is ignored. 
+        type: str
+    fingerprints:
+        description:
+            - Specifies keys to match against.
+            - Provided fingerprints will take priority over user-id "V(name) (V(comment)) <V(email)>".
+            - If O(state) is V(absent), keys with the provided fingerprints will be deleted if found 
+        type: list[str]
+    keyserver:
+        description:
+            - Specifies keyserver to upload key to.
+            - If O(state) is V(absent), this parameter will be ignored.
+        type: str
+    transient_key:
+        description:
+            - Allows key generation to use a faster, but less secure random number generator.
+        type: bool
+        default: False
+    return_fingerprints:
+        description:
+            - Allows for the return of fingerprint(s) for newly created or deleted keys(s)
```

Comment:
```suggestion
            - Allows for the return of fingerprint(s) for newly created or deleted keys(s).
```
But why not simply always return the fingerprints?

Reply:
I'm not 100% sure where the community would stand on this, but I had wanted to give users the option to keep the fingerprints hidden.

Reply:
Ended up removing the return_fingerprints since public keys naturally show the fingerprints anyways.